libr/core/disasm.c

```diff
@@ -6729,17 +6732,23 @@ R_API int r_core_disasm_pde(RCore *core, int nb_opcodes, int mode) {
 	}
 	free (buf);
 	r_reg_arena_pop (reg);
-	int len = r_list_length (ocache);
-	if (r_list_length (core->io->cache) > len) {
+	int len = r_pvector_len (&ocache);
+	if (r_pvector_len (&core->io->cache) > len) {
 		// TODO: Implement push/pop for IO.cache
 		while (len > 0) {
-			(void)r_list_pop_head (core->io->cache);
+			(void)r_pvector_pop_front (&core->io->cache);
 			len--;
 		}
-		core->io->cache->free = ocache->free;
+		core->io->cache.v.free = ocache.v.free;
 	}
 	r_io_cache_fini (core->io);
 	core->io->cache = ocache;
+	r_skyline_clear (&core->io->cache_skyline);
```

Comment:
I wonder why there is this messing around with caches in `disasm.c`. It is not exactly ideal.